Library/Homebrew/formula.rb

```diff
@@ -805,13 +805,23 @@ def skip_clean?(path)
   def link_overwrite?(path)
     # Don't overwrite files not created by Homebrew.
     return false unless path.stat.uid == File.stat(HOMEBREW_BREW_FILE).uid
-    # Don't overwrite files belong to other keg.
+     # Don't overwrite files belong to other keg except when that
+     # keg's formula is deleted.
     begin
-      Keg.for(path)
+      keg = Keg.for(path)
     rescue NotAKegError, Errno::ENOENT
       # file doesn't belong to any keg.
     else
-      return false
+      tab = Tab.for_keg(keg)
+      tap = tab.tap
+      return false if tap.nil? # this keg doesn't below to any core/tap formula, most likely coming from a DIY install.
+      begin
+        Formulary.factory("#{tap}/#{keg.name}")
+      rescue TapFormulaAmbiguityError
```

Comment:
Sorry, I miss typed in the comment. It should be `rescue FormulaUnavailableError`